### ISSUE TITLE
Clarify how tree sizes are computed

### ISSUE THREAD

**Issue by Haddox** [neherlab/SARS2-refined-fitness#3]
We compute tree sizes in a few places in the code, including:

1.  The size of the full tree used to train the GLM. The resulting tree size is used to generate the master table of per-site mutation rates by context.
2. The size of the clade subtree. This tree size is used to convert from rates to predicted counts.

Based on [this function](https://github.com/neherlab/SARS2-mut-fitness-v2/blob/aebd0e1aba02f0fc33727eca3fcbd1940c32d3d4/modules/rates.py#L187), it looks to me like we use slightly different methods for computing tree sizes in these two cases. In the first case, it looks like we compute the tree's size by summing the actual counts of synonymous mutations from the _curated_ dataframe (`train_df`). In the second case, it looks like we compute the tree's size by summing the actual counts of synonymous mutations from a _non-curated_ dataframe (`count_clade_syn`). Is that correct? If so, this seems a bit discordant to me.

@lucasesta: on slack, you said we discussed with Richard and concluded that the most consistent strategy would be to retain only the sites from the curated dataset when computing the evolutionary time for clade-specific subtrees. It should not make a significant difference since the approach we are using explicitly accounts for the potential variation in the number of sites.

I like the idea of retaining only the sites from the curated dataframe when computing the evolutionary time for clade-specific subtrees, so just wanted to double check that this is what the code is doing. And if so, it doesn't seem like we need to account for variation in the number of sites. The curated dataframe only includes the set of sites that are conserved in all founder sequences. So, if we only use those sites, the number and composition of those sites should be exactly the same, right?

**Comment by lucasesta** [comment]
@Haddox: we updated the code so to only consider the set of sites from the curated datasets when computing the evolutionary times of every clade-specific subtree. As you correctly pointed out, in this scenario the clade-specific genome composition is no longer relevant, as it is the same for all.
With this [commit](https://github.com/neherlab/SARS2-mut-fitness-v2/commit/aba15c1cd1e9b79460e4c228f059a50b526944f7), we pushed the updated outputs for the novel convention.
